### ISSUE TITLE
(GL/Vulkan) Final Fantasy VII - Mising effect during battle transition [PGXP issue]

### ISSUE THREAD

**Issue by MarcoEstevez** [libretro/beetle-psx-libretro#199]
Using mednafen_psx_hw, with openGL or vulkan, the effect (rotation transition) when you go to any battle is missing.

using software renderer and 1x IR works fine.

**Comment by MarcoEstevez** [comment]
Correct effect (software renderer) just for reference

![image](https://user-images.githubusercontent.com/3447089/27261712-e7f6f186-5448-11e7-80cb-0c723239f70a.png)

**Comment by DukeJP** [comment]
The effect works ok for me in vulkan, in 4x resolution.
But the submarine minigame crashes completely the game. 
Previous to test the effect, i updated to the last version of the core.

**Comment by MarcoEstevez** [comment]
Ill try that

**Comment by twinaphex** [comment]
For GL - you need to disable PGXP entirely.

For Vulkan - you need to disable PGXP perspective correction.

If you don't do this, the swirl won't work with these games.

**Comment by MarcoEstevez** [comment]
Thanks, Ill do as you said and will report results back

**Comment by simias** [comment]
Those effects are tricky to emulate with GL because they rely on blending the framebuffer on top of itself repeatedly. That's not something modern graphic cards are very good at.

That being said it works to a certain extent in the GL renderer, although it's clearly not accurate.

**Comment by MarcoEstevez** [comment]
Thanks for explaining, I could still reproduce the issue under GL and Vulkan with all PGXP disabled.

It plays nicelly on soft backend though :-P

**Comment by twinaphex** [comment]
A better example of this is the FF8 swirl, I think the FF7 swirl is mostly correct even with the GL renderer.

The FF8 swirl is a different story though.

The Vulkan renderer implements it correctly though. You should have a look at that and compare it with GL and soft.

**Comment by MarcoEstevez** [comment]
Hummm...

You were right, I noticed I had off - "Software framebuffer", after enabling it effect is gone

![image](https://user-images.githubusercontent.com/3447089/28733693-988cd688-73dd-11e7-81c8-ff1cc7ad0327.png)

**Comment by rz5** [comment]
@MarcoEstevez - Could you rephrase that? Does the FF7 battle swirl play correctly when the software framebuffer is ON or when it's OFF?

**Comment by MarcoEstevez** [comment]
Software FrameBuffer "On"  -->  swirl plays correctly on software, gl and Vulkan
Software FrameBuffer "Off"  -->  swirl plays correctly only on software mode, gl and vulkan shows black screen.

**Comment by rz5** [comment]
Yeah, that makes sense. That option was meant to be a speedhack in that it makes mednafen skip some work.
But after it was implemented, it didn't notice a significant increase in performance. It would be interesting if someone savvy could benchmark it.

**Comment by MarcoEstevez** [comment]
I havent noticed myself any slowdown after activating it.

**Comment by rz5** [comment]
Worth noting that there are actually 3 core options that mess with this effect:
Frame duping (speedup);
PGXP perspective correct texturing;
Software framebuffer.

**Comment by twinaphex** [comment]
I think this is finally fixed now with the GL renderer.

Let me try it briefly with Vulkan as well.

**Comment by twinaphex** [comment]
Fixed for both Vulkan and GL.

**Comment by RagnaStrife** [comment]
Can anybody check if this issue is totally and perfectly fixed on Vulkan?
I mean, in reference to this other issue: #401
